prune panic at 100%

<!--
NOTE: Not filling out the issue template needs a good reason, otherwise the
issue may be closed instantly! Please take the time to help us debugging the
problem by collecting information, even if it seems irrelevant to you. Thanks!
-->

## Output of `restic version`
restic 0.6.1 (v0.6.1-71-gbd7d5a42)
compiled with go1.8.3 on linux/amd64


## How did you start restic exactly? (Include the complete command line)
```
jean@rindvieh ~> restic -r sftp://jkahrs@10.10.10.199:2244//home/jean prune
```

## What backend/server/service did you use?
sftp

## Expected behavior
restic prunes the repository and unlocks it afterwards

## Actual behavior
the prune command paniced before the repo was unlocked

```
jean@rindvieh ~> restic -r sftp://jkahrs@10.10.10.199:2244//home/jean prune
enter password for repository: 
counting files in repo
building new index for repo
[3:53] 100.00%  21234 / 21234 packs
repository contains 21234 packs (6709929 blobs) with 93.168 GiB bytes
processed 6709929 blobs: 0 duplicate blobs, 0B duplicate
load all snapshots
find data that is still in use for 45 snapshots
```

### about 5hours later when prune is about to finish (~100%)

```
panic: runtime error: slice bounds out of range                                  goroutine 67 [running]:
main.newProgressMax.func1(0x0, 0x0, 0x0, 0x0, 0x2c, 0x0, 0xf8ddea57a12, 0xc420063e01)
	src/cmds/restic/cmd_prune.go:48 +0x354
restic.(*Progress).updateProgress(0xc476c21080, 0x0, 0x0, 0x0, 0x0, 0x2c, 0x0, 0x1)
	src/restic/progress.go:133 +0xa6
restic.(*Progress).reporter.func1()
	src/restic/progress.go:146 +0xaf
restic.(*Progress).reporter(0xc476c21080)
	src/restic/progress.go:157 +0xf5
created by restic.(*Progress).Start
	src/restic/progress.go:82 +0x151
```


## Steps to reproduce the behavior
Not sure yet. I'm trying to rerun the prune command on the repo, however this will take some time ;)
I'll update the issue later.


## Do you have any idea what may have caused this?
May be a problem with updating the progress bar?

--- a/src/cmds/restic/cmd_prune.go
+++ b/src/cmds/restic/cmd_prune.go
@@ -28,6 +28,18 @@ func init() {
 	cmdRoot.AddCommand(cmdPrune)
 }
 
+func shortenStatus(maxLength int, s string) string {
+	if len(s) <= maxLength {
+		return s
+	}
+
+	if maxLength < 3 {
+		return s[:maxLength]
+	}
+
+	return s[:maxLength-3] + "..."
+}
+
 // newProgressMax returns a progress that counts blobs.
 func newProgressMax(show bool, max uint64, description string) *restic.Progress {
 	if !show {
@@ -43,10 +55,7 @@ func newProgressMax(show bool, max uint64, description string) *restic.Progress 
 			s.Blobs, max, description)
 
 		if w := stdoutTerminalWidth(); w > 0 {
-			if len(status) > w {
-				max := w - len(status) - 4
-				status = status[:max] + "... "
-			}
+			status = shortenStatus(w, status)
 		}
 
 		PrintProgress("%s", status)

